File: agriTech/src/main/java/com/example/agriTech/mapper/device/NotificationMapper.java
package com.example.agriTech.mapper.device;

import java.time.format.DateTimeFormatter;

import org.springframework.stereotype.Component;

import com.example.agriTech.dto.sensor.NotificationDTO;
import com.example.agriTech.model.Notification;
@Component
public class NotificationMapper {
    // Định dạng thời gian chuẩn để Kotlin nhận diện được ngay
    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

    public NotificationDTO EntitytoDTO(Notification entity) {
        if(entity == null) return null;
        NotificationDTO dto = new NotificationDTO();
       dto.setId(entity.getId());
       dto.setUserId(entity.getUser().getId());
       dto.setPlantUserId(entity.getPlantUser().getId());
       // 2. Lấy deviceCode từ đối tượng Device liên kết
        if (entity.getDevice() != null) {
            dto.setDeviceCode(entity.getDevice().getDeviceCode());
        }

        // 3. Map nội dung thông báo
        dto.setTitle(entity.getTitle());
        dto.setMessage(entity.getMessage());
        dto.setRead(entity.getIsRead());
        // 4. Chuyển LocalDateTime sang String theo format chuẩn
        if (entity.getCreatedAt() != null) {
            dto.setCreatedAt(entity.getCreatedAt().format(formatter));
        }
        return dto;
    }
}

File: agriTech(DA3K)/app/src/main/java/com/example/agritechda3k/mapper/Notification/NotificationMapper.kt
package com.example.agritechda3k.mapper.Notification

import com.example.agritechda3k.api.dto.NotificationDTO
import com.example.agritechda3k.model.Notification

fun NotificationDTO.toEntityss() : Notification{
    return Notification(
        id = this.id,
        userId = this.userId,
        plantUserId = this.plantUserId,
        title = this.title,
        message = this.message,
        isRead = this.isRead,
        createdAt = this.createdAt,
        deviceCode = this.deviceCode
    )
}

fun List<NotificationDTO>.toEntityssList(): List<Notification> {
    return this.map { it.toEntityss() }
}